Sources/Orbit/Support/TextConcatenation/TextRepresentable+Concatenation.swift

```diff
@@ -0,0 +1,70 @@
+import SwiftUI
+
+/// Conctatenates two terms that can be represented as `SwiftUI.Text`
+/// - Parameters:
+///   - left: A content representable as `SwiftUI.Text`
+///   - right: A content representable as `SwiftUI.Text`
+/// - Returns: A view that is the result of concatenation of text representation of the parameters.
+///   if both paramters do not have a text representation, the returning view will produce `EmptyView`, preserving the standard Orbit behavior
+@ViewBuilder public func +(
+    left: TextRepresentable,
+    right: TextRepresentable
+) -> some View & TextRepresentable {
+    ConcatenatedText(left) + ConcatenatedText(right)
+}
+
+struct TextConcatenationPreviews: PreviewProvider {
+    static var previews: some View {
+        PreviewWrapper {
+            standalone
+            snapshot
+        }
+        .padding(.medium)
+        .previewLayout(.sizeThatFits)
+    }
+
+    static var standalone: some View {
+        (Heading("Hanoi ", style: .title1)
+             + Icon(.flightReturn, size: .xLarge)
+             + Heading(" San Pedro de Alcantara", style: .title1))
+        .lineLimit(2)
+    }
+    static var snapshot: some View {
+        VStack(alignment: .leading, spacing: .xSmall) {
+            Heading("Display Title", style: .display) + Icon(.flightReturn, size: .heading(.display))
```

Comment:
The current snapshots now use the icon size as if it would be used in a `Label` or `Tile` - but that uses specifically larger size for that case.

Suggestion: using the same size as the heading (for now using the `custom` enum, but we should add a mapping property now or in future), so that it better matches when the Icon is **inside** the text:

```
Heading(.., style) + Icon(.flightReturn, size: .custom(style.size)) + Heading(.., style)
```

Result:

<img width="331" alt="image" src="https://user-images.githubusercontent.com/35844477/186685434-688a0c8d-b32a-4602-8b71-ce563e8627e6.png">

<img width="330" alt="image" src="https://user-images.githubusercontent.com/35844477/186685775-5cbe6fd5-1e5b-4795-9ef5-95798d3d7125.png">

Reply:
Updated the snapshots